src/types/mod.rs

```diff
@@ -774,3 +902,178 @@ impl<T: Object> ValueInternal for JsFunction<T> {
+        self.args.push(arg.upcast());
+        self
+    }
+
+    /// Add multiple arguments to the arguments list.
+    pub fn args<A: Arguments<'a>>(&mut self, args: A) -> &mut Self {
+        args.append(&mut self.args);
+        self
+    }
+
+    /// Call the function as a constructor (like a JavaScript `new` expression).
+    /// If the function returns without throwing, returns the resulting object.
+    pub fn new<'b, C: Context<'b>>(&self, cx: &mut C) -> JsResult<'b, T> {
+        self.callee.do_construct(cx, &self.args)
+    }
+
+    /// Make the function call. If the function returns without throwing, the result value
+    /// is downcast to the type `V`, throwing a `TypeError` if the downcast fails.
+    pub fn call<'b: 'a, C: Context<'b>, V: Value>(&self, cx: &mut C) -> JsResult<'b, V> {
+        let undefined: Handle<JsValue> = cx.undefined().upcast();
+        let v: Handle<JsValue> = self.callee.do_call(cx, undefined, &self.args)?;
+        v.downcast_or_throw(cx)
+    }
+
+    /// Make the function call for side effect, discarding the result value. This method is
+    /// preferable to [`call()`](crate::types::Call::call) when the result value is not
+    /// needed, since it does not require specifying a result type.
+    pub fn exec<'b: 'a, C: Context<'b>>(&self, cx: &mut C) -> NeonResult<()> {
+        let undefined: Handle<JsValue> = cx.undefined().upcast();
+        let _: Handle<JsValue> = self.callee.do_call(cx, undefined, &self.args)?;
+        Ok(())
+    }
+}
+
+/// The trait for specifying arguments in a [`Call`](crate::types::Call) or
+/// [`FunctionCall`](crate::types::FunctionCall). This trait is sealed and cannot
+/// be implemented by types outside of the Neon crate.
+pub trait Arguments<'a>: ArgumentsInternal {
+    #[doc(hidden)]
+    fn append(self, args: &mut ArgsVec<'a>);
+}
+
+macro_rules! impl_arguments {
+    { (); (); } => {
+        impl ArgumentsInternal for () { }
+
+        impl<'a> Arguments<'a> for () {
+            fn append(self, _args: &mut ArgsVec<'a>) { }
+        }
+    };
+
+    { ($tname1:ident,$($tnames:ident,)*); ($vname1:ident,$($vnames:ident,)*); } => {
+        impl<'a, $tname1: Value, $($tnames: Value,)*> ArgumentsInternal for (Handle<'a, $tname1>, $(Handle<'a, $tnames>,)*) { }
+
+        impl<'a, $tname1: Value, $($tnames: Value,)*> Arguments<'a> for (Handle<'a, $tname1>, $(Handle<'a, $tnames>,)*) {
+            fn append(self, args: &mut ArgsVec<'a>) {
+                let ($vname1, $($vnames,)*) = self;
+                args.push($vname1.upcast());
+                $(args.push($vnames.upcast());)*
+            }
+        }
+
+        impl_arguments! {
+            ($($tnames,)*);
+            ($($vnames,)*);
+        }
+    };
+}
+
+impl_arguments! {
+    (V1, V2, V3, V4, V5, V6, V7, V8,
+     V9, V10, V11, V12, V13, V14, V15, V16,
+     V17, V18, V19, V20, V21, V22, V23, V24,
+     V25, V26, V27, V28, V29, V30, V31, V32,);
+
+    (v1, v2, v3, v4, v5, v6, v7, v8,
```

Comment:
It would be nice to mark most of these as `#[doc(hidden)]` to keep the docs page more manageable.

https://deploy-preview-796--neon-bindings-docs.netlify.app/neon/types/trait.arguments#foreign-impls